### DISCUSSION TITLE
infrastructure/ubiquiti/udm-dream-machine/wireguard-vpn/

### DISCUSSION THREAD

**Discussion by giscus** [fire1ce/3os.org#87] [General]
# infrastructure/ubiquiti/udm-dream-machine/wireguard-vpn/

Wireguard VPN installation and configuration guide for Ubiquiti UniFi UDM, UDM PRO, UDR and UXG

https://3os.org/infrastructure/ubiquiti/udm-dream-machine/wireguard-vpn/

**Comment by tpthaler** [comment]
Thanks for writing such a great article. I have a Dream Machine Pro (Special Edition) and I was able to successfully get WireGuard running (and listening on port 51820), but when I try to connect to it via my GL iNet Router or iPhone I just get a connection timeout. I do have a firewall rule enabled that allows ALL traffic on port 51820 in. Is there anything else I might be missing, or any advice on tools I could use to help debug my connection issue?

**Reply by fire1ce** [reply]
You should start with telnet on the your external ip and wireguard port from external wan address. It seems you can’t reach the external port of your Wan on udm.

**Reply by tpthaler** [reply]
If anyone else runs into this problem, it was due to my Ubiquiti's firmware calling WAN Local "Internet Local" and I had my firewall incorrectly configured to "Internet In"
